examples/widgets/scale/lv_example_scale_7.c

```diff
@@ -0,0 +1,76 @@
+#include "../../lv_examples.h"
+#if LV_USE_SCALE && LV_BUILD_EXAMPLES
+
+#include "../../../src/lvgl_private.h" //To expose the fields of lv_draw_task_t
+
+static void draw_event_cb(lv_event_t * e)
+{
+    lv_obj_t * obj = lv_event_get_target(e);
+    lv_draw_task_t * draw_task = lv_event_get_draw_task(e);
+    lv_draw_dsc_base_t * base_dsc = lv_draw_task_get_draw_dsc(draw_task);
+    lv_draw_label_dsc_t * label_draw_dsc = lv_draw_task_get_label_dsc(draw_task);
+    if(base_dsc->part == LV_PART_INDICATOR) {
+        if(label_draw_dsc) {
+            const lv_color_t color_idx[7] = {
+                lv_palette_main(LV_PALETTE_RED),
+                lv_palette_main(LV_PALETTE_ORANGE),
+                lv_palette_main(LV_PALETTE_YELLOW),
+                lv_palette_main(LV_PALETTE_GREEN),
+                lv_palette_main(LV_PALETTE_CYAN),
+                lv_palette_main(LV_PALETTE_BLUE),
+                lv_palette_main(LV_PALETTE_PURPLE),
+            };
+            uint8_t major_tick = lv_scale_get_major_tick_every(obj);
+            label_draw_dsc->color = color_idx[base_dsc->id1 / major_tick];
+            /* Figure out how many bytes we need to allocate */
+            char tmp_buffer[20] = {0}; /* Big enough buffer */
+            int needed_bytes = lv_snprintf(tmp_buffer, sizeof(tmp_buffer), "%.1f", base_dsc->id2 * 1.0f);
+            needed_bytes += 1U; /* null terminator */
+
+            /*Free the previously allocated text if needed*/
+            if(label_draw_dsc->text_local) lv_free((void *)label_draw_dsc->text);
+            /*Mark text_local as 1 so space for the next text is allocated and the new string is copied into it*/
```

Comment:
I just noticed that I don't understand this comment. Typo?
```suggestion
            /*Malloc the text and set text_local to make LVGL automatically free the text. (Local texts are malloced internally by LVGL. Mimic this behavior here too)*/
```

Same for the test.

Reply:
Hi, they were my own words, now it is more understandable, thanks for the feedback :smile: